### PR TITLE
Fix docs for `rejectAll`

### PR DESCRIPTION
**Pull request by melix**
Fix a copy and paste error in javadocs.

### CODE REVIEW

**Comment by melix** [comment]
@bot-gradle test QF

File: subprojects/core-api/src/main/java/org/gradle/api/artifacts/MutableVersionConstraint.java [hunk 1]
```diff
@@ -95,8 +95,7 @@
     void reject(String... versions);
 
     /**
-     * Rejects all versions of this component. Can be used to declare that a component is incompatible with another
-     * (typically, cannot have both a 2 different implementations of the same API).
+     * Rejects all versions of this component. Can be used to banish a specific component from dependency graphs.
```
Comment (big-guy) [inline subprojects/core-api/src/main/java/org/gradle/api/artifacts/MutableVersionConstraint.java:98]: Does this "banish" a component (which sounds like an exclude) or make resolution fail if that component is in the graph?

Comment (melix) [reply subprojects/core-api/src/main/java/org/gradle/api/artifacts/MutableVersionConstraint.java:98]: It makes resolution fail, I'll clarify.

**Comment by big-guy** [comment]
Thanks Cedric!

**Pull request merged**